resources/Microsoft.VSCode.Dsc/Microsoft.VSCode.Dsc.psm1

```diff
@@ -5,44 +5,81 @@ $ErrorActionPreference = 'Stop'
 Set-StrictMode -Version Latest
 
 #region Functions
-function Get-VSCodeCLIPath {
+function Search-UninstallRegistry {
+    [CmdletBinding(DefaultParameterSetName = 'User')]
     param (
-        [switch]$Insiders
-    )
-
-    # Product Codes for VSCode and VSCode Insider:
-    $vsCodeUserProductCode = '{771FD6B0-FA20-440A-A002-3B3BAC16DC50}_is1'
-    $vsCodeMachineProductCode = '{EA457B21-F73E-494C-ACAB-524FDE069978}_is1'
+        [Parameter(ParameterSetName = 'User', Mandatory = $true)]
+        [switch] $User,
 
-    $vsCodeInsidersUserProductCode = '{217B4C08-948D-4276-BFBB-BEE930AE5A2C}_is1'
-    $vsCodeInsidersMachineProductCode = '{1287CAD5-7C8D-410D-88B9-0D1EE4A83FF2}_is1'
+        [Parameter(ParameterSetName = 'Machine', Mandatory = $true)]
+        [switch] $Machine,
 
-    # Note: WOW6432 registry not checked as no uninstall entry was found.
-    $userUninstallRegistry = 'HKCU:\Software\Microsoft\Windows\CurrentVersion\Uninstall'
-    $machineUninstallRegistry = 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall'
-    $installLocationProperty = 'InstallLocation'
+        [Parameter(Mandatory = $true)]
+        [string] $DisplayName
+    )
 
-    if ($Insiders) {
-        $cmdPath = 'bin\code-insiders.cmd'
-        $insidersUserInstallLocation = TryGetRegistryValue -Key "$($userUninstallRegistry)\$($vsCodeInsidersUserProductCode)" -Property $installLocationProperty
-        if ($insidersUserInstallLocation) {
-            return $insidersUserInstallLocation + $cmdPath
+    switch ($PSCmdlet.ParameterSetName) {
+        'User' {
+            $Path = 'HKCU:\Software\Microsoft\Windows\CurrentVersion\Uninstall'
         }
+        'Machine' {
+            $Path = 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall'
+        }
+    }
 
-        $insidersMachineInstallLocation = TryGetRegistryValue -Key "$($machineUninstallRegistry)\$($vsCodeInsidersMachineProductCode)" -Property $installLocationProperty
-        if ($insidersMachineInstallLocation) {
-            return $insidersMachineInstallLocation + $cmdPath
+    $UninstallKeys = Get-ChildItem -Path $Path
+    foreach ($key in $UninstallKeys) {
+        $value = Get-ItemProperty -Path $key.PSPath
+
+        if ($value.DisplayName -eq $DisplayName) {
+            return $value
         }
-    } else {
-        $cmdPath = 'bin\code.cmd'
-        $codeUserInstallLocation = TryGetRegistryValue -Key "$($userUninstallRegistry)\$($vsCodeUserProductCode)" -Property $installLocationProperty
-        if ($codeUserInstallLocation) {
-            return $codeUserInstallLocation + $cmdPath
+    }
+}
+
+function Get-VSCodeCLIPath {
+    param (
+        [switch]$Insiders
+    )
+
+    if ($IsLinux) {
+        if ($Insiders) {
+            $InstallLocation = Join-Path ($env:PATH.Split([System.IO.Path]::PathSeparator) -match 'Microsoft VS Code Insiders') 'code-insiders'
+            if (Test-Path $InstallLocation) {
+                return $InstallLocation
+
+            }
```

Comment:
nit: You can bring this snippet outside of the enclosing If...else statement so that you only have to write this once since you're ultimately checking the result of $InstallLocation